validator/remote/src/main/java/tech/pegasys/teku/validator/remote/RemoteValidatorApiHandler.java

```diff
@@ -0,0 +1,178 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.remote;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.api.response.GetForkResponse;
+import tech.pegasys.teku.api.schema.BLSPubKey;
+import tech.pegasys.teku.api.schema.ValidatorDutiesRequest;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.bls.BLSSignature;
+import tech.pegasys.teku.datastructures.blocks.BeaconBlock;
+import tech.pegasys.teku.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.datastructures.operations.Attestation;
+import tech.pegasys.teku.datastructures.operations.SignedAggregateAndProof;
+import tech.pegasys.teku.datastructures.state.Fork;
+import tech.pegasys.teku.datastructures.state.ForkInfo;
+import tech.pegasys.teku.datastructures.validator.SubnetSubscription;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.service.serviceutils.ServiceConfig;
+import tech.pegasys.teku.validator.api.ValidatorApiChannel;
+import tech.pegasys.teku.validator.api.ValidatorDuties;
+import tech.pegasys.teku.validator.remote.apiclient.OkHttpValidatorRestApiClient;
+import tech.pegasys.teku.validator.remote.apiclient.ValidatorRestApiClient;
+
+public class RemoteValidatorApiHandler implements ValidatorApiChannel {
+
+  private final ValidatorRestApiClient apiClient;
+
+  @SuppressWarnings("unused")
+  public RemoteValidatorApiHandler(final ServiceConfig config) {
+    // TODO: create config for validator client [https://github.com/PegaSysEng/teku/issues/2450]
+    apiClient = new OkHttpValidatorRestApiClient("http://127.0.0.1:8888");
+  }
+
+  public RemoteValidatorApiHandler(final ValidatorRestApiClient apiClient) {
+    this.apiClient = apiClient;
+  }
+
+  @Override
+  public SafeFuture<Optional<ForkInfo>> getForkInfo() {
+    return SafeFuture.supplyAsync(() -> apiClient.getFork().map(this::mapGetForkResponse));
+  }
+
+  private ForkInfo mapGetForkResponse(final GetForkResponse response) {
+    final Fork fork = new Fork(response.previous_version, response.current_version, response.epoch);
+    return new ForkInfo(fork, response.genesis_validators_root);
+  }
+
+  @Override
+  public SafeFuture<Optional<List<ValidatorDuties>>> getDuties(
+      final UInt64 epoch, final Collection<BLSPublicKey> publicKeys) {
+    if (publicKeys.isEmpty()) {
+      return SafeFuture.completedFuture(Optional.of(Collections.emptyList()));
+    }
+
+    return SafeFuture.supplyAsync(
```

Comment:
I'd avoid `supplyAsync` as it doesn't give any control over which threads are used to execute the request.  Instead I'd pass in the `AsyncRunner` instance `ValidatorClient` has and use that.  Then it will use the existing thread pool for the validator client.

Or if the aim is just to ensure any exceptions get wrapped up in the returned future (good idea), you could just use `SafeFuture.of`